nifi-nar-bundles/nifi-graph-bundle/nifi-other-graph-services/src/main/java/org/apache/nifi/graph/TinkerpopClientService.java

```diff
@@ -0,0 +1,545 @@
+        "provided by processors such as the ExecuteGraphQueryRecordProcessor.")
+@SeeAlso(classNames = {"ExecuteGraphQueryRecordProcessor"})
+@RequiresInstanceClassLoading
+public class TinkerpopClientService extends AbstractControllerService implements GraphClientService {
+    public static final String NOT_SUPPORTED = "NOT_SUPPORTED";
+    private static final AllowableValue BYTECODE_SUBMISSION = new AllowableValue("bytecode-submission", "ByteCode Submission",
+            "Groovy scripts are compiled within NiFi, with the GraphTraversalSource injected as a variable 'g'. Effectively allowing " +
+                    "your logic to directly manipulates the graph without string serialization overheard."
+            );
+
+    private static final AllowableValue SCRIPT_SUBMISSION = new AllowableValue("script-submission", "Script Submission",
+            "Script is sent to the gremlin server as a submission. Similar to a rest request. "
+    );
+
+    private static final AllowableValue YAML_SETTINGS = new AllowableValue("yaml-settings", "Yaml Settings",
+            "Connection to the gremlin server will be specified via a yaml file (very flexible)");
+
+    private static final AllowableValue SERVICE_SETTINGS = new AllowableValue("service-settings", "Service-Defined Settings",
+            "Connection to the gremlin server will be specified via values on this controller (simpler). " +
+                    "Only recommended for testing and development with a simple grpah instance. ");
+
+    public static final PropertyDescriptor SUBMISSION_TYPE = new PropertyDescriptor.Builder()
+            .name("submission-type")
+            .displayName("Script Submission Type")
+            .description("A selection that toggles for between script submission or as bytecode submission")
+            .allowableValues(SCRIPT_SUBMISSION, BYTECODE_SUBMISSION)
+            .defaultValue("script-submission")
+            .required(true)
+            .build();
+
+    public static final PropertyDescriptor CONNECTION_SETTINGS = new PropertyDescriptor.Builder()
+            .name("connection-settings")
+            .displayName("Settings Specification")
+            .description("Selecting \"Service-Defined Settings\" connects using the setting on this service. Selecting \"Yaml Settings\" uses the specified yaml file for connection settings. ")
+            .allowableValues(SERVICE_SETTINGS, YAML_SETTINGS)
+            .defaultValue("service-settings")
+            .required(true)
+            .build();
+
+    public static final PropertyDescriptor CONTACT_POINTS = new PropertyDescriptor.Builder()
+            .name("tinkerpop-contact-points")
+            .displayName("Contact Points")
+            .description("A comma-separated list of hostnames or IP addresses where an OpenCypher-enabled server can be found.")
```

Comment:
References to OpenCypher should be Tinkerpop or Gremlin

Reply:
Updated to gremlin.